### PR TITLE
Merge repeated definitions

### PR DESCRIPTION
**Pull request by cjgillot**
Step forward on #66149 

I may need further context to understand the need for a separate crate.

Also, please tell me if you think of other definitions to merge.

### CODE REVIEW

File: src/librustc_mir/borrow_check/nll/type_check/mod.rs [hunk 1]
```diff
@@ -2210,38 +2210,39 @@ impl<'a, 'tcx> TypeChecker<'a, 'tcx> {
                                 }
                             }
 
-                            if let hir::MutMutable = mutability {
```
Comment (oli-obk) [inline src/librustc_mir/borrow_check/nll/type_check/mod.rs]: why did you change this to a match? A match will increase the nesting here, and for mutability it seems like there's not too much to be gained from having an exhaustive match. We're not going to extend that enum in the future with more mutability variants I believe

File: src/librustc/ty/print/pretty.rs [hunk 2]
```diff
@@ -607,10 +607,9 @@ pub trait PrettyPrinter<'tcx>:
             ty::Generator(did, substs, movability) => {
                 let upvar_tys = substs.as_generator().upvar_tys(did, self.tcx());
                 let witness = substs.as_generator().witness(did, self.tcx());
-                if movability == hir::GeneratorMovability::Movable {
-                    p!(write("[generator"));
-                } else {
-                    p!(write("[static generator"));
+                match movability {
+                  hir::Movability::Movable => p!(write("[generator")),
```
Comment (oli-obk) [inline src/librustc/ty/print/pretty.rs]: indentation is off here

File: src/librustc/hir/mod.rs [hunk 3]
```diff
@@ -1802,17 +1802,6 @@ pub struct Destination {
     pub target_id: Result<HirId, LoopIdError>,
 }
 
-/// Whether a generator contains self-references, causing it to be `!Unpin`.
```
Comment (oli-obk) [inline src/librustc/hir/mod.rs]: move the documentation on the enum and the variants to the ast type

File: src/librustc_mir/borrow_check/nll/type_check/mod.rs (hunk 1 above)
Comment (cjgillot) [reply src/librustc_mir/borrow_check/nll/type_check/mod.rs]: I tend to think exhaustive matches as less demanding in terms of brain bandwidth. However in that case, I agree and reverted the change.

**Comment by cjgillot** [comment]
Rebased with your comments.

**Pull request merged**